ledger/db/src/lib.rs

```diff
@@ -250,28 +278,49 @@ impl LedgerDB {
     /// Opens an existing Ledger Database in the given path.
     pub fn open(path: PathBuf) -> Result<LedgerDB, Error> {
         let env = Environment::new()
-            .set_max_dbs(20)
+            .set_max_dbs(21)
             .set_map_size(MAX_LMDB_FILE_SIZE)
             // TODO - needed because currently our test cloud machines have slow disks.
             .set_flags(EnvironmentFlags::NO_SYNC)
             .open(&path)?;
 
+        // Check if the database we opened is compatible with the current implementation.
+        let metadata_store = MetadataStore::new(&env)?;
+        let db_txn = env.begin_ro_txn()?;
+        let version = metadata_store.get_version(&db_txn)?;
+        global_log::info!("Ledger db is currently at version: {:?}", version);
+        db_txn.commit()?;
+
+        match version.is_compatible_with_latest() {
```

Comment:
This is the first time we've included such code, but I believe it is a nice experiment to have. I have verified it successfully upgrades a pre-existing LMDB database.

Reply:
Nice

Reply:
Interesting...